File: qwencoder-eval/instruct/eval-dev-quality/testdata/java/transpile/binarySearch/implementation/binary_search.rb
def binary_search(a, x)
	index = -1

	min = 0
	max = a.length - 1

	while index == -1 && min <= max
		m = (min + max) / 2

		if x == a[m]
			index = m
		elsif x < a[m]
			max = m - 1
		else
			min = m + 1
		end
	end

	return index
end

File: qwencoder-eval/instruct/eval-dev-quality/testdata/ruby/transpile/binarySearch/implementation/binarySearch.go
package light

func binarySearch(a []int, x int) int {
	index := -1

	min := 0
	max := len(a) - 1

	for index == -1 && min <= max {
		m := (min + max) / 2

		if x == a[m] {
			index = m
		} else if x < a[m] {
			max = m - 1
		} else {
			min = m + 1
		}
	}

	return index
}
